pkg/scheduler/preemption/fairsharing/target.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fairsharing
+
+import (
+	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+// TargetClusterQueue is a ClusterQueue which yields candidate
+// workloads for preemption.
+type TargetClusterQueue struct {
+	ordering *TargetClusterQueueOrdering
+	targetCq *cache.ClusterQueueSnapshot
+}
+
+// InClusterQueuePreemption indicates that the TargetClusterQueue is
+// the preemptor ClusterQueue; i.e. the preemptor ClusterQueue is
+// considering its own workloads for priority based preemption.
+func (t *TargetClusterQueue) InClusterQueuePreemption() bool {
+	return t.targetCq == t.ordering.preemptorCq
+}
+
+func (t *TargetClusterQueue) PopWorkload() *workload.Info {
+	head := t.ordering.clusterQueueToTarget[t.targetCq.GetName()][0]
+	t.ordering.clusterQueueToTarget[t.targetCq.GetName()] = t.ordering.clusterQueueToTarget[t.targetCq.GetName()][1:]
+	return head
+}
```

Comment:
Can we extract `t.targetCq.GetName()` to a variable?

Reply:
Done.